falwa/f90_modules/compute_qref_and_fawa_first.f90

```diff
@@ -26,7 +26,8 @@ SUBROUTINE compute_qref_and_fawa_first(pv, uu, vort, pt, tn0, imax, JMAX, kmax,
   REAL :: qbar(nd,kmax)
 
   pi = acos(-1.)
-  dp = pi/float(jmax-1)
+  dp = pi/float(jmax-1)  !!!  This is dlat
+  dl = 2*pi/float(imax-1)  !!!! pragallva- correction added on Dec 13, 2023. This is dlon
```

Comment:
# 2. Changes in the fortran code

Thanks for taking care of both the modules in `QGFieldNH18` and `QGFieldNHN22`! The changes look good to me. There is one spot which I worry that may have inconsistency though - in `QGFieldBase`, `self.dphi` and `self.dlambda` is actually precomputed:

https://github.com/csyhuang/hn2016_falwa/blob/46b77ab3a1c0ef614bc9bdeb354ce749b17f02b2/falwa/oopinterface.py#L157-L158

which will be used downstream to calculate the flux convergence. Even though computing it once more in Fortran shall not take too much time (+they shall be consistent with `dp` and `dl` in Fortran code), passing them in from python code to Fortran module will guarantee consistency (I missed that when putting together the code before). That would require changing the interface of the Fortran module though. Let me know if you are interested in doing that. If not, I can also do that later after merging your pull request! Let me know what you prefer!

Reply:
Hi Clare, I saw in several places dp (or dphi) is defined inside several of the fortran codes. I have followed the same syntax to define dl (or dlambda). Theoretically they look consistent to me in both python and fortran scripts. I'd prefer not to make this change myself else I will mess something up.  I will let you take care of this. ;)